src/script_opt/ScriptOpt.cc

```diff
@@ -578,4 +578,23 @@ void analyze_scripts()
 	analyze_scripts_for_ZAM(pfs);
 	}
 
+void profile_script_execution()
+	{
+	if ( analysis_options.profile_ZAM )
+		{
+		report_ZOP_profile();
+
+		for ( auto& f : funcs )
+			{
+			if ( f.Body()->Tag() == STMT_ZAM )
+				cast_intrusive<ZBody>(f.Body())->ProfileExecution();
+			}
+		}
+	}
+
+void finish_script_execution()
+	{
+	profile_script_execution();
```

Comment:
Are you intending for this to do more in the future? If not, can this function go away and zeek-setup can just call `profile_script_execution()`?

Reply:
I don't currently picture it doing anything more.  It just seems a bit strange to me that `zeek-setup` would know about ZAM script profiling, rather than the more generic notion of "let script optimization know that things are done".  The current name is at the same semantic level as `analyze_scripts()` (the function above `finish_script_execution()`).

So yeah a matter of taste ...